dandi/metadata.py

```diff
@@ -963,4 +913,14 @@ def get_generator(start_time: datetime, end_time: datetime) -> models.Activity:
 
 
 def prepare_metadata(metadata: dict) -> Dict[str, Any]:
+    """
+    Convert "flatdata" [1]_ for an asset into raw [2]_ "schemadata" [3]_
+
+    .. [1] a flat `dict` mapping strings to strings & other primitive types;
+       returned by `get_metadata()`
```

Comment:
I wondered if we are to unify should  we just then rename the `get_metadata` into `get_flatdata` and deprecate `get_metadata` to make it all consistent... but then I realized that `get_flatdata` excluding the context of "metadata" is somewhat misleading and `get_flatdata_metadata` is odd.  So may be we should make it `flatmetadata` and `get_flatmetadata` or better even `adhocmetadata` and `get_adhocmetadata`.... let's sleep on that... for now good and we could do such massive RF later ...